plugins/coverbrowser.koplugin/mosaicmenu.lua

```diff
@@ -648,6 +651,44 @@ function MosaicMenuItem:update()
                 self.has_description = true
             end
 
+            if not self.menu.cover_info_cache then
+                self.menu.cover_info_cache = {}
+            end
+
+            -- Current page / pages are available or more accurate in .sdr/metadata.lua
+            -- We use a cache (cleaned at end of this browsing session) to store
+            -- page, percent read and book status from sidecar files, to avoid
+            -- re-parsing them when re-rendering a visited page
+            if not self.menu.cover_info_cache then
+                self.menu.cover_info_cache = {}
+            end
+            local percent_finished, status
+            local pages = bookinfo.pages -- default to those in bookinfo db
+            if DocSettings:hasSidecarFile(self.filepath) then
+                self.been_opened = true
+                if self.menu.cover_info_cache[self.filepath] then
+                    percent_finished, status = unpack(self.menu.cover_info_cache[self.filepath])
```

Comment:
I think you removed the first item because of "unused variable", but now, percent_finished will be the saved nb of page, status will be the saved percent_finished, no ?
There are tricks for select the 2nd++ items from unpack, but cleanest to just tag the unused variable so luacheck don't complain:
`local pages -- luacheck: no unused`

Although this might not even be the case here, this feels like a too quick and wrong removal:
```diff
-                    pages, percent_finished, status = unpack(self.menu.cover_info_cache[self.filepath])
+                    percent_finished, status = unpack(self.menu.cover_info_cache[self.filepath])
```

Reply:
I don't really need progress for pages. And, I removed it only when I get it from the cache, not elsewhere. That is I'm storing it to cache like `listmenu.lua`. I thought `unpack` unpacks by key like standard object deconstruction.

Anyway, restoring this.

Reply:
> I thought unpack unpacks by key like standard object deconstruction.

Nope, all these are plain arrays of 3 elements with no key/name.